[CI] Make the linux-modules-extras conditional

zdohnal opened the pull request:
Several runs fail on sound dummy part, check if this works...

jamacku commented on .github/workflows/ci.yml:
```diff
@@ -180,24 +180,26 @@ jobs:
-      - name: Set up snd-dummy
+      - name: Check if linux-modules-extra is available
+        id: check-snd-dummy
         if: (!(contains(matrix.extra, 'unittests') || contains(matrix.extra, 'vimtags')))
```
You can try this condition:

```suggestion
        if: (!(contains(matrix.extra, 'unittests') || contains(matrix.extra, 'vimtags'))) && runner.name != 'ubuntu-22.04'
```

[runner context](https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context)

zdohnal committed:
ci.yml: Make installation of linux-modules-extra conditional

linux-modules-extra package are not available on Ubuntu 22 at the
moment, which breaks CI runs. The change make its installation
conditional as the original `if-else` structure tried to do.
